src/cgame/cg_rocket_datasource.cpp

```diff
@@ -1137,7 +1156,20 @@ static void CG_Rocket_CleanUpMapList( const char* )
 
 static void CG_Rocket_SetMapListIndex( const char*, int index )
 {
-	rocketInfo.data.mapIndex = index;
+	if ( index >= 0 && index < int( rocketInfo.data.mapList.size() ) )
```

Comment:
if we're checking, let's also warn when the else case happens

Reply:
Well, this part was cargo-cult copy-paste, but I added the warning.

Reply:
Well, that's not a good idea, the game prints this warning at startup:

```
Warn: Map list index -1 is out of range [0, 0[ 
Warn: Map list index -1 is out of range [0, 0[ 
Warn: Map list index -1 is out of range [0, 0[ 
```

Reply:
so before we were setting mapIndex to -1 and everything was working ok? Does this mean with our current code, we cannot "unselect" a map? -1 means no selection

Reply:
I assume we cannot unselect a map but I don't see that as an issue, emptying `g_defaultMap` will select the first one, one cannot create a server without a map anyway. It also makes the UI consistent by always displaying the levelshot, meaning the position of the UI is always consistent: I can center it vertically with a levelshot displayed and expect it to always look good.